innersource-activity-score.md

```diff
@@ -0,0 +1,92 @@
+
+## Sketch (optional)
+
+```//TODO: add sketch here ;-)```
+
+## Solutions
+
+The InnerSource activity score can be derived automatically from common parameters like GitHub stars, watches, and forks. In addition, it considers activity parameters like last update and creation date of the repo to give young projects with a lot of traction a boost.
+Projects with contributing guidelines and issues (public backlog) may get a higher ranking.
+
+All of this can be fetched and calculated automatically using the GitHub API (see [reference implementation](TODO) below). The code below assumes the variable `repo` contains an entity fetched from the GitHub API. Manual adjustments can be made on top if needed.
+
+``` javascript
+// calculate a virtual InnerSource score from stars, watches, commits, and issues
+function calculateScore(repo) {
+    // weighting:
+    // forks and watches count most, then stars, add some little score for open issues, too
+    let iScore = 1 + repo["forks_count"] * 5 + repo["watchers_count"] + repo["stargazers_count"] / 3 + repo["open_issues_count"] / 5;
+    let iDaysSinceLastUpdate = (new Date().getTime() - new Date(repo.updated_at).getTime()) / 1000 / 86400;
+    // updated at this month: adds a bonus between 0..1 to overall score (1 = updated today, 0 = updated more than 100 days ago)
+    iScore = iScore * (1 + (100 - Math.min(iDaysSinceLastUpdate, 100)) / 100);
+    // new repos (created in the last 365 days) that have been updated recently (in the last 30 days) get a boost of up to 1000
+    let iDaysSinceCreation = (new Date().getTime() - new Date(repo.created_at).getTime()) / 1000 / 86400;
+    let iBoost = (1000 - Math.min(iDaysSinceLastUpdate, 100) * 10);
+    iScore += (iDaysSinceCreation < 365 ? (iDaysSinceLastUpdate < 30 ? iBoost : 0) : 0);
+    // give projects with contribution guidelines (CONTRIBUTING.md) file a static boost of 100
+    iScore += (repo["_InnerSourceMetadata"] && repo["_InnerSourceMetadata"]["guidelines"] ? 100 : 0);
+    // final score is a rounded value starting from 0
+    iScore = Math.round(iScore - 1);
+    // add score to metadata on the fly
+    repo._InnerSourceMetadata = repo._InnerSourceMetadata || {};
+    repo._InnerSourceMetadata.score = iScore;
+    return iScore;
+}
+```
+
+## Resulting Context
+
+Projects can be sorted and presented by the InnerSource activity score to give a meaningful order in a list or portal. The activity score can be calculated on the fly or in a background job that evaluates all projects on a regular basis and stores a list of results.
+
+A crawler that regularly searches all InnerSource repositories (e.g. tagged with a certain label in an enterprise GitHub instance) can be a helpful solution as well. It provides a ranked list of projects that can be used as an input for tools like the [InnerSource Portal](innersource-portal.md), a search engine, or an interactive chat bot. 
```

Comment:
What do you mean by "enterprise GitHub instance"? Same as "GitHub organization"?

Reply:
The resulting context might also indicate the impact of not including the soft KPIs (e.g., the ranking might not be as meaningful in certain aspects, which might later point to another pattern that would work in the soft KPIs).

Reply:
reworked this part, an enterprise GitHub instance is a self-hosted GitHub instance inside the company, similar to a paid plan on GitHub where you can host private repos

Reply:
will leave out the soft KPI part here, i moved it to the solution and the caveats are described in forces